src/dmd/dsymbolsem.d

```diff
@@ -3890,7 +3890,9 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
         Linterfaces:
             bool foundVtblMatch = false;
 
-            foreach (b; cd.interfaces)
+            BaseClasses bases;
+            cd.allInterfaces(bases);
+            foreach (b; bases)
```

Comment:
What appears to be happening is an array is constructed of all the interfaces. Then, this array is iterated over. The array is a temporary array.

Memory allocation can be avoided by having allInterfaces accept a delegate that operates on each interface found. The delegate would be the body of the foreach loop.

Reply:
I have changed it so it takes a delegate. I now changed the `foreach` to a function call which passes a delegate, but I could probably change it so it would still work with a `foreach.`